src/node/interfaces.cpp

```diff
@@ -183,21 +185,25 @@ class NodeImpl : public Node
     int getNumBlocks() override
     {
         LOCK(::cs_main);
-        assert(std::addressof(::ChainActive()) == std::addressof(m_context->chainman->ActiveChain()));
-        return m_context->chainman->ActiveChain().Height();
+        assert(std::addressof(::ChainActive()) == std::addressof(chainman().ActiveChain()));
+        return chainman().ActiveChain().Height();
     }
     uint256 getBestBlockHash() override
     {
-        assert(std::addressof(::ChainActive()) == std::addressof(m_context->chainman->ActiveChain()));
-        const CBlockIndex* tip = WITH_LOCK(::cs_main, return m_context->chainman->ActiveChain().Tip());
+        const CBlockIndex* tip;
+        {
+            LOCK(cs_main);
+            assert(std::addressof(::ChainActive()) == std::addressof(chainman().ActiveChain()));
```

Comment:
In commits "node/ifaces: NodeImpl: Use an accessor for ChainMan" (3460a07d88d841c009e7b7efc0be2ae89bb4a63e)
and "node/ifaces: ChainImpl: Use an accessor for ChainMan" (3b75ec491a68eb6fb137c2f829e7dde93b3c3189)

Not important unless there will be a lot more of these, but I think it would be better not to add nested scopes for temporary asserts that we are going to want to remove and collapse later. The asserts seem good in other places where they can be removed with scripted diffs just deleting the assert lines. But places like this should be easier to clean up later using a helper like:

```c++
//! Temporary accessor that can be removed after
//! https://github.com/bitcoin/bitcoin/pull/20158, but helps check correctness
//! of refactored code in the meantime.
static inline CChain& ActiveChain(ChainstateManager& chainman) EXCLUSIVE_LOCKS_REQUIRED(cs_main)
{
    assert(std::addressof(::ChainActive()) == std::addressof(chainman.ActiveChain()));
    return chainman.ActiveChain();
}
```

that could substituted out with a scripted-diff without needing to change surrounding code. Could also be appealing if you think usage looks nicer:

```diff
-        assert(std::addressof(::ChainActive()) == std::addressof(chainman().ActiveChain()));
-        return chainman().ActiveChain().Height();
+        return ActiveChain(chainman()).Height();
```